docs/src/meshes/p4est_mesh.md

```diff
@@ -599,4 +601,343 @@ A simple mesh file, which is used also in `examples/p4est_3d_dgsem/elixir_euler_
+
+Then, a different balance function could look like this:
+```julia
+function balance_p4est_multirate!(mesh::ParallelP4estMesh, dg, cache,
+                                  # Vector{Vector{Int}} = Multirate levels with associated element indices
+                                  level_info_elements, 
+                                  # Vector{Int} = RHS Runge-Kutta stage evaluations per multirate level
+                                  stages)
+    rhs_per_element = zeros(Int, nelements(dg, cache))
+
+    # Loop over different multirate levels
+    for level in eachindex(level_info_elements)
+        # Access computational cost = RHS evaluations for this multirate level
+        rhs_evals = stages[level]
+
+        # Loop over all elements on this multirate level
+        for element in level_info_elements[level]
+            rhs_per_element[element] = rhs_evals
+        end
+    end
+
+    # Copy `rhs_per_element` to the `p4est` mesh backend
+    iter_volume_c = cfunction(save_rhs_evals_iter_volume, Val(ndims(mesh)))
+    iterate_p4est(mesh.p4est, rhs_per_element; iter_volume_c = iter_volume_c)
+
+    # Set up according weighting function that can now be evaluated since
+    # the user data has been copied to the mesh backend
+    weight_fn_c = cfunction(weight_fn_multirate, Val(ndims(mesh)))
+    # Perform mesh partitioning with custom weighting function
+    partition!(mesh; weight_fn = weight_fn_c)
+end
+```
+This redistributes the mesh among the MPI ranks.
+Now, the same needs to be done for the solver (recall that mesh and solver are quite decoupled for `p4est` meshes).
+Thus, the call to [`partition!`](https://github.com/trixi-framework/Trixi.jl/blob/f76ccaf23a1df150f9a185f0faa770ebc1abc417/src/meshes/p4est_mesh.jl#L2189-L2197) needs to be followed by a call to [`rebalance_solver!`](https://github.com/trixi-framework/Trixi.jl/blob/f76ccaf23a1df150f9a185f0faa770ebc1abc417/src/callbacks_step/amr_dg.jl#L8-L92) to rebalance the solver data.
+Overall, this non-standard partitioning can be realized as
+```julia
+# Get cell distribution for standard partitioning
+global_first_quadrant = unsafe_wrap(Array,
+                                    unsafe_load(mesh.p4est).global_first_quadrant,
+                                    mpi_nranks() + 1)
+# Need to copy `global_first_quadrant` to different variable as the former will change 
+# due to the call to `partition!`
+old_global_first_quadrant = copy(global_first_quadrant)
+
+# Get (global) element distribution to accordingly balance the solver
+level_info_elements = some_user_defined_function()
+
+# Balance such that each rank has the same number of RHS calls                                    
+balance_p4est_multirate!(mesh, dg, cache, level_info_elements, alg.stages)
+# Actual move of elements across ranks
+rebalance_solver!(u0, mesh, equations, dg, cache, old_global_first_quadrant)
+reinitialize_boundaries!(semi.boundary_conditions, cache) # Needs to be called after `rebalance_solver!`
+```
```

Comment:
Maybe I'm missing something, but I am not quite sure, where this could should live. Can you maybe clarify that a bit?

Reply:
I would place this in the `resize!` of an integrator I think.

Reply:
Would you mind adding a sentence explaining that very briefly? Because otherwise, I think, as a reader it would be difficult to, where to place this code.